### DISCUSSION TITLE
第Ⅱ部でなんでテストツール作るかについて

### DISCUSSION THREAD

**Discussion by rinahira** [code-polaris/reading-circle-tdd#82] [General]
第Ⅱ部、xUnitどう使うか？って話なのかと思いきや、思いっきり自分でテスティングフレームワーク作ってみる話でびっくりしました。。
ですが、下記記事を読んだらだいぶ腹落ちしました。

テスト駆動開発の第2章は"なぜxUnitなのか"？
https://qiita.com/kotauchisunsun/items/2919a7833c90d188e0d1

xUnit自体ケントベックさんが自分で作ったものなので、読者も作ってみれば思想が分かるよ！ってことだろうし
今やテスティングフレームワークがあることが当たり前になってしまっていますが、ケントさんが本を書いた時はそうではないので「自分でテスティングフレームワーク作れるよ！」ってことを示したかったんですかねー。

**Comment by shino51** [comment]
私もそれびっくりしました！
急に言語が変わるのも、Javaで書こうとすると複雑になりすぎるからなんでしょうね。実際、私は第19章のsetUp methodを作るところで、これは複雑になりすぎるな…と思って既存のsetUp methodを使ってしまいました。悔しいですね。
